src/lustre/lustreTypeChecker.ml

```diff
@@ -1749,169 +1775,58 @@ let rec type_check_group: tc_context -> LA.t ->  unit tc_result list
-                         ^^ "Constant and type context \n"
-                         ^^ "TC Context\n%a\n"
-                         ^^"===============================================\n")
-          pp_print_tc_context global_ctx
-
-        (* Inline constants in constant declarations *)
-        ; Log.log L_trace ("===============================================\nStarting inlining constants") 
-        ; (try (inline_constants global_ctx sorted_tys_consts) with
-          | Out_of_bounds (pos, err) -> error pos err) >>= fun (global_ctx', inlined_cs) ->  
-            
-          Log.log L_trace ( "Inlining constants done: \nConstant decls: %a"
-                            ^^"\n===============================================\n")
-            LA.pp_print_program inlined_cs
-
-          (* sort nodes and contract decls to make sure there are no recursive node/contract calls *)
-          ; AD.sort_declarations node_and_contract_decls >>= fun sorted_node_contract_decls -> 
-            
-            (* gather the typing context from nodes/functions and contracts *)
-            tc_context_of global_ctx' sorted_node_contract_decls >>= fun tc_ctx ->
-            Log.log L_trace ("===============================================\n"
-                             ^^ "Phase 1: Completed Building TC Global Context\n"
-                             ^^ "TC Context\n%a\n"
-                             ^^"===============================================\n")
-              pp_print_tc_context tc_ctx
-            ; (* type check the nodes and contract decls using this base typing context  *)
-              let tc_res = (type_check_decl_grps tc_ctx [prg]) in
-              Log.log L_trace ("===============================================\n"
-                               ^^ "Phase 2: Type checking declaration Groups Done\n"
-                               ^^"===============================================\n")                  
-              ; report_tc_result tc_res
+(**************************************************************************************
+ * The main function of the file that kicks off type checking or type inference flow  *
+ ***************************************************************************************)
+
+let type_check_infer_program: constants_or_nodes -> tc_context -> LA.t -> tc_context tc_result
+  = fun c_or_n ctx prg ->
+  match c_or_n with
+  | Constants_and_types ->
+     (Log.log L_trace ("===============================================\n"
+                       ^^ "Building TC Global Context\n"
+                       ^^"===============================================\n")
+     (* Build base constant and type context *)
+     ; build_type_and_const_context ctx prg >>= fun global_ctx ->
+       (* Inline constants in constant declarations *)
+       Log.log L_trace ("===============================================\nStarting inlining constants") 
+       ; let (global_ctx', inlined_cs) = inline_constants global_ctx prg in
```

Comment:
You should explain the difference between this `inline_constants` function and the one called from `LustreInput` in line 133.

Reply:
I have removed this function. So we no longer need it.